### ISSUE TITLE
Add bash as dependency?

### ISSUE THREAD

**Issue by deekayen** [droboports/crashplan#24]
My 4.5.0 install didn't auto-upgrade, then I saw bash is a new dependency in #21. When I cat /tmp/DroboApps/crashplan/depends.txt, it still only lists locale and java. Adding bash sounds like an appropriate change.

**Comment by ricardopadilha** [comment]
I have given up on trying to chase the dependencies of the auto-update mechanism of Crashplan. Every release since 4.5.0 has changed the procedure in some way that breaks everything I have done so far. Worse yet, their update scripts do some really unsafe things, and do not follow bash best practices (e.g. no `set -o nounset` followed by some very liberal use of `rm -fr`).

So I have decided that the best policy for the Drobos is to prevent the auto-update mechanism completely. This will have two advantages: (1) it will prevent bad surprises caused by the update script, and (2) it will allow me to have enough time to roll out a decently tested version of the app.

As it currently stands every time there is an update I'm running against the clock to package the latest version, test it on my own Drobos (putting my data at risk), and then quickly push it out. This is no way to do proper QA.

Since I've always released a new version of the package when a new version comes out, the end result will be the same for the end users. It's a win-win in my book.

TL;DR: I'm currently investigating a way to prevent the auto-update completely, thus eliminating the need for all the workarounds currently in place to make the (constantly changing) update procedure work.

By the way, the latest version (4.7.0) is currently available from Drobo Dashboard. I have a custom version of 4.7.0 running on my 5N with the patches to prevent auto-upgrade, just waiting for the next upgrade to hit. It is a very simple change, which hopefully won't have side-effects.

**Comment by scdemanett** [comment]
This sounds like the best and most reasonable approach to me. The app will continue to operate and backup until an update hits the Dashboard and the end user experience should be much better. Unfortunate that Code42 operates in this manner and the last thing I need is an unchecked rm -fr going off. You do fantastic work Ricardo and we VERY much appreciate it, thank you!

**Comment by dmzettl** [comment]
I can only second this. Thanks for your great work Ricardo!

Sent from my iPad

On 24 May 2016, at 17:02, scdemanett <notifications@github.com> wrote:

This sounds like the best and most reasonable approach to me. The app will
continue to operate and backup until an update hits the Dashboard and the
end user experience should be much better. Unfortunate that Code42 operates
in this manner and the last thing I need is an unchecked rm -fr going off.
You do fantastic work Ricardo and we VERY much appreciate it, thank you!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub
<https://github.com/droboports/crashplan/issues/24#issuecomment-221299542>
